### ISSUE TITLE
PushNotification is undefined when use PhoneGap Build

### ISSUE THREAD

**Issue by dland512** [phonegap/phonegap-plugin-push#2201]
### Expected Behaviour
Should be able to use this plugin with PhoneGap Build.

### Actual Behaviour
The PushNotification variable is not defined.

### Reproduce Scenario (including but not limited to)
Registering for push token. Everything work fine on my local dev box but when use PhoneGap Build it doesn't work.

#### Steps to Reproduce 
1. Point PhoneGap Build to [this git repo](https://github.com/dland512/cordovapush.git).
2. Build the app (you won't be able to without the dev cert and provisioning profile).
3. Download the app to an iOS device.
4. Open the app. It cannot register for a push token because PushNotification is undefined.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 7

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
None (yet)

#### Cordova CLI version and cordova platform version
    cordova --version                                    # e.g. 6.0.0
    7.1.0

    cordova platform version android                     # e.g. 4.1.1

    Installed platforms:
       ios 4.5.4
    Available platforms: 
       android ~6.3.0

#### Plugin version
    cordova plugin version | grep phonegap-plugin-push   # e.g. 1.5.3
    phonegap-plugin-push 2.1.3 "PushPlugin"

#### Sample Push Data Payload
    {
        "aps": {
            "alert":"Testing",
            "badge":1,
            "sound":"default"
         }
    }

#### Sample Code that illustrates the problem
```
var app = {
    // Application Constructor
    initialize: function() {
        document.addEventListener('deviceready', this.onDeviceReady.bind(this), false);
    },

    // deviceready Event Handler
    //
    // Bind any cordova events here. Common events are:
    // 'pause', 'resume', etc.
    onDeviceReady: function() {
        this.receivedEvent('deviceready');
    },

    // Update DOM on a Received Event
    receivedEvent: function(id) {
        console.log('device is ready');

        var parentElement = document.getElementById(id);
        var listeningElement = parentElement.querySelector('.listening');
        var receivedElement = parentElement.querySelector('.received');

        listeningElement.setAttribute('style', 'display:none;');
        receivedElement.setAttribute('style', 'display:block;');

        try {
            var pn = PushNotification;
            console.log('PushNotification is defined');
        }
        catch (e) {
            console.log('PushNotification is not defined');
        }
        const push = PushNotification.init({
            ios: {
                alert: 'true',
                badge: 'true',
                sound: 'true'
            }
        });

        push.on('registration', function(data) {
            console.log('registered');
            console.log(JSON.stringify(data, null, 4));
        });

        push.on('notification', function(data) {
            console.log('notified');
            console.log(JSON.stringify(data, null, 4));
        });

        push.on('error', function(e) {
            console.log('error');
            console.log(e.toString());
        });
    }
};

app.initialize();
```

#### Logs taken while reproducing problem

> device is ready
> PushNotification is not defined

**Comment by macdonst** [comment]
@dland512 the PGB team has asked for people with issues regarding phonegap build please start an issue on the support forum: https://forums.adobe.com/community/phonegap/build

Please include an app id of an app that can replicate this issue.

Also, the plugin no longer supports iOS7.

**Comment by dland512** [comment]
Thanks for the info. I will ask them. Also the iOS version was a type. It's actually 11.2.2.

**Comment by jcesarmobile** [comment]
You have to set the PGB CLI version to 7.1.0 or 8.0.0, if you don't set it it will use 6.5.0, which is not compatible.
Check my sample app
https://github.com/jcesarmobile/phonegap-plugin-push-sample

**Comment by dland512** [comment]
Yep! That's exactly right @jcesarmobile . I just noticed that in the PGB build log:

```
Installing "phonegap-plugin-push" at "2.1.3" for ios
Plugin doesn't support this project's cordova-ios version. cordova-ios: 4.3.1, failed version requirement: >=4.5.0
Skipping 'phonegap-plugin-push' for ios
```

As [this page](https://build.phonegap.com/current-support) states, the phonegap version defaults to `cli-6.5.0` which uses iOS phonegap 4.3.1, and this plugin [requires iOS 4.4.0 or higher](https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/INSTALLATION.md#installation-requirements). Once I saw that it was quite clear. I'm a little surprised that build passes when plugins cannot be installed, but whatever.

Anyway, the fix was to add the following line to to `config.xml`:

`<preference name="phonegap-version" value="cli-6.5.0" />`

After that building on PGB worked fine.
